internal/clients/composer/client.go

```diff
@@ -107,67 +87,33 @@ func (cc *ComposerClient) request(method, url string, headers map[string]string,
 		req.Header.Add(k, v)
 	}
 
-	token := func() string {
-		cc.tokenMu.RLock()
-		defer cc.tokenMu.RUnlock()
-		return cc.accessToken
+	token, err := cc.tokener.Token(req.Context())
+	if err != nil {
+		return nil, err
 	}
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token()))
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
 
 	resp, err := cc.client.Do(req)
 	if err != nil {
 		return nil, err
 	}
 
-	if resp.StatusCode == http.StatusUnauthorized {
-		err = cc.refreshToken()
+	if resp.StatusCode == http.StatusForbidden || resp.StatusCode == http.StatusUnauthorized {
+		token, err = cc.tokener.ForceRefresh(req.Context())
+		if err != nil {
+			return nil, err
+		}
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+		_, err = body.Seek(0, io.SeekStart)
```

Comment:
`body` could be `nil` here, so add a `if body != nil` check before seeking. If it's a get request, there's no body.